tracer/build/_build/CodeGenerators/CallTargetsGenerator.cs

```diff
@@ -0,0 +1,743 @@
+            {
+                return attribute.AttributeType.FullName.Equals("Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.AdoNetClientInstrumentMethodsAttribute");
+            }
+
+            static bool IsTargetSignatureAttribute(Mono.Cecil.CustomAttribute attribute)
+            {
+                return attribute.AttributeType.FullName.Equals("Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.AdoNetClientInstrumentMethodsAttribute/AdoNetTargetSignatureAttribute");
+            }
+
+            static AdoNetSignature GetAdoNetSignature(TypeDefinition type, CustomAttribute attribute)
+            {
+                var hasMisconfiguredInput = false;
+
+                string methodName = null;
+                string returnTypeName = null;
+                string[] parameterTypeNames = null;
+                int? integrationKind = null;
+                int? returnType = null;
+                string callTargetType = null;
+
+                foreach (var namedArgument in attribute.Properties)
+                {
+                    switch (namedArgument.Name)
+                    {
+                        case nameof(AdoNetSignatureAttributeProperties.MethodName):
+                            methodName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(AdoNetSignatureAttributeProperties.ReturnTypeName):
+                            returnTypeName = namedArgument.Argument.Value?.ToString();
+                            break;
+                        case nameof(AdoNetSignatureAttributeProperties.ParameterTypeNames):
+                            parameterTypeNames = GetStringArray(namedArgument.Argument.Value);
+                            break;
+                        case nameof(AdoNetSignatureAttributeProperties.CallTargetType):
+                            callTargetType = ((TypeDefinition)namedArgument.Argument.Value).FullName;
+                            break;
+                        case nameof(AdoNetSignatureAttributeProperties.CallTargetIntegrationKind):
+                            integrationKind = namedArgument.Argument.Value as int?;
+                            break;
+                        case nameof(AdoNetSignatureAttributeProperties.ReturnType):
+                            returnType = namedArgument.Argument.Value as int?;
+                            break;
+                        default:
+                            hasMisconfiguredInput = true;
+                            break;
+                    }
+
+                    if (hasMisconfiguredInput)
+                    {
+                        break;
+                    }
+                }
+
+                return new AdoNetSignature(
+                    className: type.FullName,
+                    targetMethodName: methodName!,
+                    targetReturnType: returnTypeName,
```

Comment:
As above, IMO, add some validation for invalid values

Reply:
Done